packages/dds/shared-object-base/src/sharedObject.ts

```diff
@@ -888,7 +888,7 @@ export interface SharedObjectKind<out TSharedObject = unknown>
  * See {@link @fluidframework/fluid-static#ContainerSchema} for how this is used in the declarative API.
  * @internal
  */
-export function createSharedObjectKind<TSharedObject>(
+export function createSharedObjectKind<TSharedObject extends IFluidLoadable>(
```

Comment:
Since IFluidLoadable ended up in the public API, this is fine (Originally I wasn't clear on where that would land), but I don't think this adds any value from a safety perspective. IChannelFactory already ensures this type constraint is met by the internal types so this is safe if I understand correctly, even without your change.

The change simply prevents a user of createSharedObjectKind from type erasing the fact the their underlying implementation implements IFluidLoadable when specifying the type parameter manually.

This removes existing functionality, and I don't think has a safety gain. As this is an internal API and we don't need that functionality, its fine, but I'm not sure its beneficial.

Of course if we keep the change to SharedObjectKind, then this is required.